### PR TITLE
[GSoC 2024]: Improvements to Tracker - Implementing an Incremental Search - Final Report upload

### PR DESCRIPTION
**Pull request by Calisto-Mathias**
This commit contains the Hugo Server upload for my Haiku Project. Although, I was not able to meet all of the expecations, I was able to set up a core edition of the project up and running. I am currently making some last minute changes after which those will be uploaded to gerrit as well :)

### CODE REVIEW

**Comment by diversys** [comment]
GSoC 2023 -> 2024

**Comment by humdingerb** [comment]
When you change the 2023 -> 2024, also put the title of your project in there.
Otherwise, LGTM. @nielx ?

**Comment by Calisto-Mathias** [comment]
Sorry for the typo in the commit message. I didn't notice it. I amended the commit now as well as added the title of the project in the PR as well as the commit message.

File: content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+type = "blog"
+title = "[GSoC 2024]: Implementing Incremental Search [Final Report]"
+author = "calisto-mathias"
+date = "2024-08-24 01:46:45+05:30"
+tags = ["haiku", "software", "gsoc", "gsoc-24", "tracker", "user-interface"]
++++
+
+## Introduction
+
+The goal of this post is to document the changes I’ve successfully made during the GSoC period, the current state of the project, future enhancement goals, and a few other topics. I also want to extend my thanks to the Haiku developers and community for the opportunity to work on this fantastic operating system. :)
```
Comment (nielx) [inline content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md:11]: Suggest to remove this, I doubt this renders nicely on the website, as well as I do not think it is consistent with the rest of our style to put emoticons on the snippets on the homepage.

File: content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+## Background
+
+Haiku is a feature-rich operating system known for its incredibly fast search system. This efficiency is largely due to its powerful Query system and the Be file system's effective use of file metadata, which provides superior indexing compared to many other operating systems.
```
Comment (nielx) [inline content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md:15]: I like the enthusiasm, but I don't think you have the evidence to back up that it is fast and superior to other operating systems. In fact, I suspect that due to all the investments Apple and Microsoft have made in their search capabilities, they are miles ahead in terms of performance. I'd rather rephrase it to emphasizing that Haiku's predecessor BeOS made metadata a primary characteristic of the filesystem, and introduced file system queries at the kernel level, which opens up a very interesting way to explore search.

File: content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+<figure>
+	<img src="/files/blog/calisto-mathias/Final-Report/directories-entries-in-menu.png" alt="New Find Panel Directory Selector" width="598px">
+	<center><figcaption>Selecting Directories in the New Find Panel (contd.)</figcaption></center>
+</figure>
+
+### Adding the Design for an Incremental Search Find Panel
+
+The most crucial and challenging aspect of my project was implementing the design for the incremental search find panel, which associates attribute columns with their corresponding search columns. While I managed to implement the core functionality, there are a few limitations that need to be addressed in the future.
+
+Currently, the incremental find panel integrated into the Find Panel results window is under review on GitHub (and soon on Gerrit). It will take some time before it is merged into the main codebase. Despite the challenges, I managed to get the core functionality working, allowing users to pause and restart their queries, as well as adjust search terms, all within the results window without switching back and forth repeatedly.
```
Comment (nielx) [inline content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md:73]: Currently under review on Gerrit, right? You can link to the PR

File: content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+## What’s Still Left to Do?
+
+Implementing this project in one go was quite challenging. Initially, I hoped to complete everything by the project deadline, but that proved unrealistic. It took me some time to grasp the technical details involved. However, I believe I can complete the remaining work with a bit more time and by tackling it in smaller steps.
+
+Here’s what remains to be done in the incremental find panel:
+
+1. **Incremental Search with Keystrokes:** The goal was to refresh results with each keystroke. Although I implemented this, it turned out to be unstable due to the high load from frequent clearing, refreshing, and restarting of queries. I had planned to develop a debouncer to manage this load but didn’t manage to complete it. This will be my primary focus after the GSoC’24 period ends.
```
Comment (nielx) [inline content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md:91]: Maybe expand this section a little bit on which strategies you would like to explore, so that in case you are not able to finish it due to other obligations, others will have a head start on doing this.

**Review by nielx — COMMENTED** [review]
I added some suggestions/things to think about.

File: content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md (hunk 4 above)
Comment (Calisto-Mathias) [reply content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md]: Yup. I wrote down what I was thinking of doing. Thanks :)

File: content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md (hunk 3 above)
Comment (Calisto-Mathias) [reply content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md]: Yup. I had linked it at the bottom. I also added it as a link in that section now. I hope that's fine.

File: content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md (hunk 1 above)
Comment (Calisto-Mathias) [reply content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md]: Oh okay. I removed the emoticon.

File: content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md (hunk 2 above)
Comment (Calisto-Mathias) [reply content/blog/calisto-mathias/2024-08-24_[gsoc_2024]_implementing_incremental_search_[final_report].md]: Yup, I rewrote this in the newer commit.

**Comment by Calisto-Mathias** [comment]
I made all the required changes in this commit @nielx  . I hope all of them are fine :D
